lib/Model/Purchase/Purchase.php

```diff
@@ -0,0 +1,323 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Fazland\FattureInCloud\Model\Purchase;
+
+use Fazland\FattureInCloud\Client\ClientInterface;
+use Fazland\FattureInCloud\Exception\NotFoundException;
+use Fazland\FattureInCloud\Util\Json;
+
+/**
+ * @property string $id
+ * @property \DateTime $date
+ * @property \DateTime $expireNext
+ */
+class Purchase implements \JsonSerializable
+{
+    const PURCHASE = 'acquisti';
+
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    public $idSupplier;
+
+    /**
+     * @var string
+     */
+    public $invoiceNumber;
+
+    /**
+     * @var string
+     */
+    public $accrualDate;
+
+    /**
+     * @var string
+     */
+    public $name;
+
+    /**
+     * @var \DateTime
+     */
+    private $date;
+
+    /**
+     * @var string
+     */
+    public $category;
+
+    /**
+     * @var string
+     */
+    public $netAmount;
+
+    /**
+     * @var string
+     */
+    public $vatAmount;
+
+    /**
+     * @var string
+     */
+    public $withholdingTax;
+
+    /**
+     * @var string
+     */
+    public $retirementTax;
+
+    /**
+     * @var int
+     */
+    public $deductibleTax;
+
+    /**
+     * @var int
+     */
+    public $deductibleVat;
+
+    /**
+     * @var string
+     */
+    public $depreciation;
+
+    /**
+     * @var string
+     */
+    public $costCentre;
+
+    /**
+     * @var string
+     */
+    public $totalAmount;
+
+    /**
+     * @var string
+     */
+    public $currency;
+
+    /**
+     * @var string
+     */
+    public $currencyChange;
+
+    /**
+     * @var string
+     */
+    public $description;
+
+    /**
+     * @var string
+     */
+    public $fileAttached;
+
+    /**
+     * @var string
+     */
+    public $linkAttached;
+
+    /**
+     * @var \DateTime
+     */
+    private $expireNext;
+
+    /**
+     * @var string
+     */
+    public $type;
+
+    /**
+     * @var bool
+     */
+    public $payed;
+
+    /**
+     * @var array|Payment[]
+     */
+    public $listPayments = [];
+
+    /**
+     * The original data, as fetched from the APIs.
+     *
+     * @var array
+     */
+    private $originalData;
+    /**
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __get($name)
+    {
+        switch ($name) {
+            case 'id':
+                return $this->id;
+                break;
+            case 'date':
+                return $this->date;
+                break;
+            case 'expireNext':
+                return $this->expireNext;
+                break;
+            default:
+                throw new \Error('Undefined property "'.$name.'"');
+        }
+    }
+
+    public function __set($name, $value)
+    {
+        if (null !== $value) {
+            switch ($name) {
+                case 'date':
+                case 'expireNext':
+                    $value = str_replace('/', '-', $value);
```

Comment:
Same here: these values should be instances of \DateTime(Immutable)

Reply:
like above